doc/exts/pylint_messages.py

```diff
@@ -62,28 +59,62 @@ def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
     initialize_extensions(linter)
 
 
-def _get_message_data(data_path: Path) -> tuple[str, str, str, str]:
-    """Get the message data from the specified path."""
+def _get_example_code(data_path: Path) -> str:
+    """Get the example code from the specified path."""
+    if not data_path.exists():
+        raise AssertionError(
+            f"Documentation examples path {data_path} does not exist. "
+            "Please create it and add an example."
+        )
+
     good_py_path = data_path / "good.py"
     bad_py_path = data_path / "bad.py"
     details_rst_path = data_path / "details.rst"
     related_rst_path = data_path / "related.rst"
-    if not data_path.exists():
-        _create_placeholders(data_path, details_rst_path, good_py_path)
+
     good_code = _get_titled_rst(
         title="Correct code", text=_get_python_code_as_rst(good_py_path)
     )
-    bad_code = _get_titled_rst(
-        title="Problematic code", text=_get_python_code_as_rst(bad_py_path)
-    )
+    # Handle multi-file bad code templates
+    if bad_py_path.exists():
+        bad_code = _get_titled_rst(
+            title="Problematic code", text=_get_python_code_as_rst(bad_py_path)
+        )
+    elif (data_path / "bad").exists():
+        bad_files: list[str] = []
+        for file in (data_path / "bad").iterdir():
+            if file.suffix == ".py":
+                bad_files.append(
+                    f"""\
+``{file.name}``:
+
+.. literalinclude:: /{file.relative_to(Path.cwd())}
+    :language: python
+
+"""
+                )
+        bad_code = _get_titled_rst(title="Problematic code", text="".join(bad_files))
+    else:
+        bad_code = ""
```

Comment:
We should create a generic function for this as we also need to handle ``good/`` directories with multiple file imo. I can do it, it's a lot easier with pycharm than vscode.

Reply:
Yeah I didn't want to add code that we don't actively use right now, but go ahead and add it! Whatever makes it easier to complete this process the quickest!